deepspeed/utils/timer.py

```diff
@@ -37,7 +37,7 @@ def stop(self):
             """Stop the timer."""
             assert self.started_, 'timer is not started'
             torch.cuda.synchronize()
-            self.elapsed_ += (time.time() - self.start_time)
+            self.elapsed_ = (time.time() - self.start_time)
```

Comment:
Is there a reason for changing this from accumulation? There is a reset api for explicitly resetting `self.eplased_`.

Reply:
I would like to have the previous captured elapsed time saved somewhere and it seems the accumulation of self.elapsed_ is not used anywhere and the timers get reset when a new iteration starts. I can also use a new field in the timer to record past timings.

Reply:
I see. The nvme offloading uses accumulation property to time asynchronous operations such as [here](https://github.com/microsoft/DeepSpeed/blob/1f82ab78c2206b97f2d27041da8b7b127180e8e0/deepspeed/runtime/swap_tensor/async_swapper.py#L140).  If you need a reset on stop(), perhaps you can add a reset_flag (default false) to the api?

Reply:
That works well. I reverted the change to the elapsed_ accumulation and added the reset flag to timer stop() https://github.com/cli99/DeepSpeed/blob/fix-flops-profiler/deepspeed/utils/timer.py#L36.